### ISSUE TITLE
Syntax Rule Issues (0.8)

### ISSUE THREAD

**Issue by MeirionHughes** [MeirionHughes/aurelia-template-lint#53]
Syntax Rule and checking of static type usage is still opt-in; Any bugsor false-negatives within 0.8 version please report here. Please create a separate Issue for feature requests (including expanding the syntax rule's support range). 

* [x] unhandled kind errors

**Comment by zakjan** [comment]
Properties of arrays of items of imported types are not supported.

item.ts
```
export interface Item {}
```

page.ts
```
import {Item} from "./item";

export class PageViewModel {
    items: Item[];
}
```

page.html
```
<template>
    ${items.length}
</template>
```

lint output
```
[10:35:52] WARNING: cannot find 'length' in type 'Item' Ln 3 Col 5 /src/page.html
```

**Comment by zakjan** [comment]
There are some `unhandled kind` errors in my lint output. Can you make them show where they were found?

```
unhandled kind FunctionType in resolveTypeName
unhandled kind AnyKeyword in resolveTypeName
unhandled kind TypeLiteral in resolveTypeName
```

**Comment by MeirionHughes** [comment]
I should be able to fix those; its basically the type resolver doesn't have cases for those `SyntaxKind` values.

**Comment by MeirionHughes** [comment]
I've patched out the console logging so it should hopefully not pollute your output; I've added some triage tests to make sure those field types don't cause error outputs for now.

**Comment by atsu85** [comment]
Hi,
After updating to `aurelia-template-lint@0.8.2` i get lots of errors related to `@bindable` fields
> ERROR: cannot find 'x' in type 'Field'
```html
${field.x}
```
```TypeScript
import {autoinject, bindable} from "aurelia-framework";
import {Field} from "./field";
@autoinject
export class AFieldDetails {
    @bindable field: Field;
}
```

Installation details:
* Win 8
* Node v6.3.0
* npm 3.10.3
* template-lint version:
```shell
>npm ls
....
+-- gulp-aurelia-template-lint@0.8.0
| `-- aurelia-template-lint@0.8.2
|   +-- aurelia-binding@1.0.0-rc.1.0.2
|   | +-- aurelia-logging@1.0.0-rc.1.0.0
|   | +-- aurelia-metadata@1.0.0-rc.1.0.0
|   | +-- aurelia-pal@1.0.0-rc.1.0.0
|   | `-- aurelia-task-queue@1.0.0-rc.1.0.0
|   +-- aurelia-dependency-injection@1.0.0-rc.1.0.0
|   +-- aurelia-pal-nodejs@0.1.5
|   | `-- jsdom@9.4.1
|   |   +-- acorn@2.7.0
|   |   +-- array-equal@1.0.0
|   |   +-- webidl-conversions@3.0.1
|   |   `-- whatwg-url@3.0.0
|   |     `-- webidl-conversions@3.0.1
|   +-- aurelia-polyfills@1.0.0-rc.1.0.0
|   +-- aurelia-templating@1.0.0-rc.1.0.0
|   | +-- aurelia-loader@1.0.0-rc.1.0.0
|   | `-- aurelia-path@1.0.0-rc.1.0.0
|   +-- aurelia-templating-binding@1.0.0-rc.1.0.0
|   +-- glob@7.0.5
|   | `-- minimatch@3.0.2
|   +-- parse5@2.1.5
|   `-- template-lint@0.7.7
|     `-- parse5@2.1.5
...
```

**Comment by atsu85** [comment]
Issue with repeater:

> ERROR: cannot find 'row' in type 'ANavbar'

```html
<li repeat.for="row of router.navigation">
    <a href.bind="row.href">${row.title}</a>
</li>
```
```TypeScript
...
@autoinject
export class ANavbar {
    @bindable router: Router;
...
```

**Comment by MeirionHughes** [comment]
That last one should be fixed; try 0.8.3. I've updated the test for the bindable, but cannot reproduce. See if it works without the bindable? as it might be a different issue. 

Closing this in favour of individual issues per problem.

**Comment by atsu85** [comment]
@MeirionHughes, 
regarding my [first issue](https://github.com/MeirionHughes/aurelia-template-lint/issues/53#issuecomment-231203344)

> try 0.8.3. I've updated the test for the bindable, but cannot reproduce

I suspected incorrectly, that the problem was related to `@bindable` fields. What mattered was how the member in another class (`Field`) was declared. I created separate issue #61 for that and also sent a [PR](https://github.com/MeirionHughes/aurelia-template-lint/pull/62) with failing test and fix for it :)
